crates/services/consensus_module/poa/src/signer.rs

```diff
@@ -90,8 +102,40 @@ async fn sign_with_kms(
     let sig = k256::ecdsa::Signature::from_der(&signature_der)
         .map_err(|_| anyhow!("invalid DER signature from AWS KMS"))?;
     let sig = sig.normalize_s().unwrap_or(sig);
-    let sig_bytes = <[u8; 64]>::from(sig.to_bytes());
-    Ok(Signature::from_bytes(sig_bytes))
+
+    // This is a hack to get the recovery id. The signature should be normalized
+    // before computing the recovery id, but aws kms doesn't support this, and
+    // instead always computes the recovery id from non-normalized signature.
+    // So instead the recovery id is determined by checking which variant matches
+    // the original public key.
+
+    let recid1 = RecoveryId::new(false, false);
+    let recid2 = RecoveryId::new(true, false);
```

Comment:
Why are we only testing two out of four possible recovery ID:s? Shouldn't this be
```suggestion
    let recid1 = RecoveryId::new(false, false);
    let recid2 = RecoveryId::new(true, false);
    let recid3 = RecoveryId::new(true, true);
    let recid4 = RecoveryId::new(false, true);
```

Reply:
I see the assertion "reduced-x form coordinates are never generated by this", but I don't understand why we can assume that this is the case. Is that a guarantee from KMS?

Reply:
Most implementations, e.g. ethereum, disallow reduced-x form signatures. See [EIP-155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md). While it's true that reduced-x signatures can be genenerated, it's exceedingly rare, approximately once per 2e36 signatures. Still, it's likely not smart to panic here and we should indeed return an error so that the caller can retry. However, no retry logic is currently implemented, so the distinction doesn't matter right now. Still feels like it makes sense to fix this, at least the panic message, so I'll do so.

Reply:
Also, the current fuel-crypto interface pretends to be infallible, and will indeed panic on such signatures.

Reply:
Thanks for the explanation. Yeah I'm not a fan of hidden panics even if they are unlikely.

Reply:
Unlikely is an understatement here: This is less likely than winning lottery five times in a row. I wouldn't usually prepare for uuid collisions either, and this is way less likely.

Fixed in 8ed3df7ea.

Reply:
Fair enough.